common/archive.cpp

```diff
@@ -63,16 +63,29 @@ int Archive::listMatchingMembers(ArchiveMemberList &list, const Path &pattern) c
 
 SeekableReadStream *MemcachingCaseInsensitiveArchive::createReadStreamForMember(const Path &path) const {
 	String translated = translatePath(path);
+	bool isNew = false;
 	if (!_cache.contains(translated)) {
 		_cache[translated] = readContentsForPath(translated);
+		isNew = true;
 	}
 
-	const SharedArchiveContents& entry = _cache[translated];
+	SharedArchiveContents* entry = &_cache[translated];
 
-	if (entry.isFileMissing())
+	if (!entry->makeStrong()) {
```

Comment:
We need comments here.

Reply:
Done

Reply:
Umm, did you forget to push or something? I was meaning that we need comments to the developers explaining what does "strong" mean here

Reply:
Now I get it. Let me add it.